spec/relativetimeformat.html

```diff
@@ -124,22 +124,16 @@ <h1>MakePartsList ( _pattern_, _unit_, _parts_ )</h1>
         </pre>
       </p>
       <emu-alg>
+        1. Let _patternParts_ be PartitionPattern(_pattern_).
         1. Let _result_ be a new empty List.
-        1. Let _beginIndex_ be ! Call(%StringProto_indexOf%, _pattern_, « "{", *0* »).
-        1. Let _length_ be the number of elements in _pattern_.
-        1. Repeat, while _beginIndex_ is an integer index into _pattern_
-          1. Set _endIndex_ to ! Call(%StringProto_indexOf%, _pattern_, « "}", _beginIndex_ »).
-          1. Assert: _endIndex_ is not -1, otherwise the pattern would be malformed.
-          1. If _beginIndex_ is greater than _nextIndex_, then
-            1. Let _literal_ be a substring of _pattern_ from position _nextIndex_, inclusive, to position _beginIndex_, exclusive.
-            1. Add new part Record { [[Type]]: `"literal"`, [[Value]]: _literal_ } as a new element of the list _result_.
-          1. Let _p_ be the substring of _pattern_ from position _beginIndex_, exclusive, to position _endIndex_, exclusive.
-          1. Assert: _p_ is `"0"`.
-          1. For each _part_ in _parts_, do
-            1. Add new part Record { [[Type]]: _part_.[[Type]], [[Value]]: _part_.[[Value]], [[Unit]]: _unit_ } as a new element on the List _result_.
-        1. If _nextIndex_ is less than _length_, then
-          1. Let _literal_ be the substring of _pattern_ from position _nextIndex_, exclusive, to position _length_, exclusive.
-          1. Add new part Record { [[Type]]: `"literal"`, [[Value]]: _literal_ } as a new element of the list _result_.
+        1. For each element _patternPart_ of _patternParts_, in List order, do
+          1. Let _part_ be _patternPart_.[[Type]].
+          1. If _part_ is `"literal"`, then
+            1. Append Record { [[Type]]: `"literal"`, [[Value]]: _patternPart_.[[Value]] } to _result_.
+          1. Else,
+            1. Assert: _part_ is `"0"`
+            1. For each _part_ in _parts_, do
```

Comment:
We should probably rename one of the _part_ variables here. (Or inline `_patternPart_.[[Type]]` in the two places it's used.)